File: packages/client/ios/RNOta/Resolver/BundleResolution.swift
import Foundation

/// Result of one synchronous resolution pass (M1).
public struct BundleResolution: Equatable, Sendable {
  public var source: BundleSource
  /// Absolute filesystem path when ``source`` is ``BundleSource/ota``; otherwise `nil`.
  public var bundlePath: String?
  public var slotId: String?
  public var reason: ResolutionReason

  public static func embedded(reason: ResolutionReason) -> BundleResolution {
    BundleResolution(source: .embedded, bundlePath: nil, slotId: nil, reason: reason)
  }

  public static func ota(absolutePath: String, slotId: String) -> BundleResolution {
    BundleResolution(
      source: .ota,
      bundlePath: absolutePath,
      slotId: slotId,
      reason: .activeBundle
    )
  }
}

File: packages/client/android/src/main/java/com/rnota/resolver/BundleResolution.kt
package com.rnota.resolver

/**
 * Result of one synchronous resolution pass (M1).
 *
 * When [source] is [BundleSource.EMBEDDED], [bundlePath] is null — the host app
 * loads its packaged asset. When [source] is [BundleSource.OTA], [bundlePath] is
 * an absolute filesystem path to `bundle.hbc`.
 */
data class BundleResolution(
  val source: BundleSource,
  val bundlePath: String?,
  val slotId: String?,
  val reason: ResolutionReason,
) {
  companion object {
    fun embedded(reason: ResolutionReason): BundleResolution =
      BundleResolution(
        source = BundleSource.EMBEDDED,
        bundlePath = null,
        slotId = null,
        reason = reason,
      )

    fun ota(
      absolutePath: String,
      slotId: String,
    ): BundleResolution =
      BundleResolution(
        source = BundleSource.OTA,
        bundlePath = absolutePath,
        slotId = slotId,
        reason = ResolutionReason.ACTIVE_BUNDLE,
      )
  }
}
